File: src/features/housing/mock/housing.ts
import type { Housing } from "@/types/housing";

export const mockHousingListings: Housing[] = [
  {
    id: "1",
    owner_id: "user-1",

    title: "Modern 2BHK Near University",
    description:
      "Fully furnished 2BHK apartment with good connectivity and nearby facilities.",

    rent: 18000,
    deposit: 36000,

    city: "Delhi",
    locality: "North Campus",
    address: "University Road, Delhi",

    latitude: 28.6139,
    longitude: 77.2090,

    house_type: "2BHK",
    sharing_type: "Private",
    gender_preference: "Any",

    available_from: "2026-08-01",

    is_furnished: true,
    available: true,

    verified: true,

    contact_number: "9876543210",

    images: [
      "https://images.unsplash.com/photo-1560448204-e02f11c3d0e2",
    ],

    created_at: "2026-07-23",
    updated_at: "2026-07-23",
  },


  {
    id: "2",
    owner_id: "user-2",

    title: "Affordable PG for Students",
    description:
      "Clean PG with meals, WiFi and easy transport access.",

    rent: 9000,
    deposit: 10000,

    city: "Bangalore",
    locality: "Koramangala",
    address: "5th Block Koramangala",

    latitude: 12.9352,
    longitude: 77.6245,

    house_type: "PG",
    sharing_type: "Shared",
    gender_preference: "Female",

    available_from: "2026-08-15",

    is_furnished: true,
    available: true,

    verified: true,

    contact_number: "9123456789",

    images: [
      "https://images.unsplash.com/photo-1522708323590-d24dbb6b0267",
    ],

    created_at: "2026-07-23",
    updated_at: "2026-07-23",
  },
];

export const mockHousing = mockHousingListings;

File: backend/models/housing.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    Boolean,
    Date,
    DateTime,
    ForeignKey,
    Numeric,
    JSON,
)
from sqlalchemy.sql import func

from database.base import Base


class Housing(Base):
    __tablename__ = "housing"

    id = Column(Integer, primary_key=True, index=True)

    owner_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False,
    )

    title = Column(String(150), nullable=False)

    description = Column(Text, nullable=False)

    rent = Column(Numeric(10, 2), nullable=False)

    deposit = Column(Numeric(10, 2), nullable=False)

    city = Column(String(100), nullable=False)

    locality = Column(String(150), nullable=False)

    address = Column(Text, nullable=False)

    house_type = Column(String(50), nullable=False)

    sharing_type = Column(String(50), nullable=False)

    gender_preference = Column(String(30), nullable=False)

    is_furnished = Column(Boolean, default=False)

    available_from = Column(Date, nullable=False)

    available = Column(Boolean, default=True)

    verified = Column(Boolean, default=False)

    contact_number = Column(String(20), nullable=False)

    images = Column(JSON, default=list)

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
    )

    updated_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
    )